coders/fits.c

```diff
@@ -145,27 +145,34 @@ static MagickBooleanType IsFITS(const unsigned char *magick,const size_t length)
 
 static inline double GetFITSPixel(Image *image,int bits_per_pixel)
 {
+  double pixel = 0;
   switch (image->depth >> 3)
   {
     case 1:
-      return((double) ReadBlobByte(image));
+      pixel=((double) ReadBlobByte(image));
     case 2:
-      return((double) ((short) ReadBlobShort(image)));
+      pixel=((double) ((short) ReadBlobShort(image)));
     case 4:
     {
       if (bits_per_pixel > 0)
-        return((double) ReadBlobSignedLong(image));
-      return((double) ReadBlobFloat(image));
+        pixel=((double) ReadBlobSignedLong(image));
+      pixel=((double) ReadBlobFloat(image));
     }
     case 8:
     {
       if (bits_per_pixel > 0)
-        return((double) ((MagickOffsetType) ReadBlobLongLong(image)));
+        pixel=((double) ((MagickOffsetType) ReadBlobLongLong(image)));
     }
     default:
       break;
   }
-  return(ReadBlobDouble(image));
+  pixel=(ReadBlobDouble(image));
+
+  // Override any undefined pixels to black.
+  if (isnan(pixel))
```

Comment:
This should use the `IsNaN` macro instead and I am wonder if we should only do this when call to `ReadBlobDouble` is made.

Reply:
The float could technically come back as a NaN too.
I expect any of the integer values casted to double will return false for NaN, so the override to 0 shouldn't affect them.

Reply:
Ok, the logic here got weird. I changed to only do the NaN check when using ReadBlobDouble like you suggested. That seems like the simplest solution